handler_test.go

```diff
@@ -123,6 +123,36 @@ func TestUpdateStar(t *testing.T) {
 	}
 }
 
+func TestUpdateStarDoesNotDuplicate(t *testing.T) {
+	var m mock.Mock
+	currentUser := bson.NewObjectId()
+	m.On("One", &item{}).Return(nil).Run(func(args mock.Arguments) {
+		*args.Get(0).(*item) = item{ID: "stable/wordpress", StargazersIDs: []bson.ObjectId{currentUser}}
+	})
+	dbSession = testutil.NewMockSession(&m)
+	oldGetCurrentUserID := getCurrentUserID
+	getCurrentUserID = func(_ *http.Request) (bson.ObjectId, error) { return currentUser, nil }
+	defer func() { getCurrentUserID = oldGetCurrentUserID }()
+	tests := []struct {
+		name        string
+		requestBody string
+		wantCode    int
+	}{
+		{"noop", `{"id": "stable/wordpress", "has_starred": true}`, http.StatusOK},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.wantCode == http.StatusOK {
+				m.On("UpdateId", "stable/wordpress", bson.M{"$push": bson.M{"stargazers_ids": currentUser}})
```

Comment:
We don't want UpdateId to be called right? I think we can remove this and also use https://godoc.org/github.com/stretchr/testify/mock#Mock.AssertNotCalled to ensure it wasn't called.